backend/api/models.py

```diff
@@ -0,0 +1,59 @@
+from pydantic import BaseModel
```

Comment:
This is an implementation choice I'm mostly raising for discussion to see what folks think about the pattern we want to use for this codebase: Right now we have 10 models defined but I believe only 4 unique. As we add endpoints, that gap may diverge further (or not :man_shrugging: ). There's a kind of tradeoff here in terms of explicitly creating models for each endpoint vs using shared classes where possible. There's also a middle ground where we can DRY things up like:

```python
class BaseUser(BaseModel):
    email: str
    is_manager: bool
    is_admin: bool


class PostSessionResponse(BaseUser):
    pass


class GetUsersResponse(BaseUser):
    id: int


class GetCurrentUserResponse(BaseUser):
    pass


class PostUserRequest(BaseUser):
    pass


class PostUserResponse(BaseUser):
    id: int


class PutUserRequest(BaseUser):
    id: int


class PutUserResponse(BaseUser):
    id: int
```

Reply:
Good point regarding DRYing things up via the `BaseUser`. It felt a little silly making so many models that were essentially identical. I am okay with the above, but I am leaning a little more towards shared classes where possible. For example:

```python
class BaseUser(BaseModel):
    email: str
    is_manager: bool
    is_admin: bool
    
class IntResponse(BaseUser):
    id: int

class IntRequest(BaseUser):
    id: int
    
class PassResponse(BaseUser):
    pass
    
class PassRequest(BaseUser):
    pass
```

My suggested names are terrible, but I wanted to get the point across. I guess if we notice that the shape of the requests/responses are evolving we could diverge from a shared class model.

Reply:
I'm not sure I fully understand the above. I think with the removal of the deleted responses, if we wanted to consolidate to shared classes + DRY things up, we could do something like:

```python
class GoogleLoginData(BaseModel):
    token: str


class BaseUser(BaseModel):
    email: str
    is_manager: bool
    is_admin: bool


class FullUser(BaseUser):
    id: int
```

Then the endpoints would just need to be modified to use the appropriate types as applicable.

Reply:
That makes sense - I was still keeping with the theme of having separate response / request classes. This is a further consolidation. I prefer the above as opposed to the middle ground.